File: problems/392-Is-Subsequence/solution.c
bool isSubsequence(char * s, char * t){
    int sLen = strlen(s), tLen = strlen(t);
    
    if(sLen > tLen) {
        return false;
    }
    
    int sIndex = 0, tIndex = 0;
    
    while(sIndex < sLen && tIndex < tLen) {
        while (tIndex < tLen && s[sIndex] != t[tIndex]) {
            tIndex++;
        }
        if (tIndex < tLen && s[sIndex] == t[tIndex]) {
            sIndex++;
            tIndex++;
        }
    }
    
    if(sIndex == sLen) {
        return true;
    } else {
        return false;
    }
}

File: problems/392-Is-Subsequence/solution.go
func isSubsequence(s string, t string) bool {
    sIdx, tIdx := 0, 0
    for sIdx < len(s) && tIdx < len(t) {
        for tIdx < len(t) {
            if s[sIdx] != t[tIdx] {
                tIdx++
            } else {
                tIdx++
                sIdx++
                break
            }
        }
    }

    if sIdx < len(s) {
        return false
    } else {
        return true
    }
}

